github/github.go

```diff
@@ -880,6 +880,32 @@ func cloneRequest(r *http.Request) *http.Request {
 	return r2
 }
 
+// formatRateLimitResetDuration formats d to look like "[rate reset in 2s]" or
+// "[rate reset in 87m02s]" for the positive durations. And like "[rate limit was reset 87m02s ago]"
+// for the negative cases.
+func formatRateReset(d time.Duration) string {
+	isNegative := d < 0
```

Comment:
> Please let me know if you disagree.

In my personal opinion, I find code that uses variables for format string (to printf calls) harder to read than code that uses separate paths and constant format strings. I find having a `isNegative` boolean which does what it says more clear.

When looking at `return fmt.Sprintf(fmtStr, timeString)`, it's hard to immediately see what it'll print (without having to execute an entire block of code in your head). The following is much more clear to me:

```Go
if isNegative {
	return fmt.Sprintf("[rate limit was reset %v ago]", timeString)
}
return fmt.Sprintf("[rate reset in %v]", timeString)
```